itou/api/data_inclusion_api/tests.py

```diff
@@ -72,36 +72,120 @@ def test_list_structures(self):
+                    "antenne": False,
                     "lien_source": f"http://testserver{reverse('siaes_views:card', kwargs={'siae_id': siae.pk})}",
-                },
-                {
-                    "id": str(antenne.uid),
-                    "typologie": antenne.kind.value,
-                    "nom": antenne.display_name,
-                    "siret": antenne.siret,
-                    "rna": "",
-                    "presentation_resume": "",
-                    "presentation_detail": "",
-                    "site_web": antenne.website,
-                    "telephone": antenne.phone,
-                    "courriel": antenne.email,
-                    "code_postal": antenne.post_code,
-                    "code_insee": "",
-                    "commune": antenne.city,
-                    "adresse": antenne.address_line_1,
-                    "complement_adresse": antenne.address_line_2,
-                    "longitude": antenne.longitude,
-                    "latitude": antenne.latitude,
-                    "source": antenne.source,
-                    "date_maj": _str_with_tz(antenne.updated_at),
-                    # Antenne references parent structure
-                    "structure_parente": str(siae.uid),
-                    "lien_source": f"http://testserver{reverse('siaes_views:card', kwargs={'siae_id': antenne.pk})}",
-                },
+                    "horaires_ouverture": "",
+                    "accessibilite": "",
+                    "labels_nationaux": [],
+                    "labels_autres": [],
+                    "thematiques": [],
+                }
             ],
         )
 
+    def test_list_structures_antenne_with_user_created_with_proper_siret(self):
+        siae_1 = SiaeFactory(siret="10000000000001")
+        siae_2 = SiaeFactory(siret="10000000000002", convention=siae_1.convention)
+        siae_3 = SiaeFactory(siret="10000000000003", source=Siae.SOURCE_USER_CREATED, convention=siae_1.convention)
+
+        response = self.authenticated_client.get(
+            self.url,
+            format="json",
+            data={"type": "siae"},
+        )
+        self.assertEqual(response.status_code, 200)
+
+        structure_data_list = response.json()["results"]
+
+        for siae, siret, antenne in [
+            (siae_1, siae_1.siret, False),
+            (siae_2, siae_2.siret, False),
+            # siae is user created, but it has its own siret
+            # so it is not an antenne according to data.inclusion
+            (siae_3, siae_3.siret, False),
+        ]:
+            structure_data = next(
+                (structure_data for structure_data in structure_data_list if structure_data["id"] == str(siae.uid)),
+                None,
+            )
+            with self.subTest(siret=siae.siret):
+                assert structure_data is not None
+                assert structure_data["siret"] == siret
+                assert structure_data["antenne"] == antenne
```

Comment:
c'est hyper complexe ce `next` avec un tuple comprehension, boucle for et subtest. Tu veux pas tester directement genre
```
assert data[0]["siret"] == siae1.siret
...
```
Ca sera beaucoup plus lisible et surtout tu testeras l'ordre de ta liste, qui devrait etre fixé. Sinon c'est un bug et tu devrais forcer l'ordre de sortie de ton API.

Reply:
* ça n'est pas mon avis mais je te change ça ! ;)
* d'expérience je trouve que tester l'ordre un peu partout est moins maintenable et diffuse la responsabilité de vérifier ce point.

Reply:
bon ok ça n'est pas plus mal :)